app/models/procedure_revision.rb

```diff
@@ -412,6 +413,13 @@ def header_sections_are_valid?
     repetition_tdcs_errors + root_tdcs_errors
   end
 
+  def expressions_regulieres_are_valid?
+    types_de_champ_public.to_a
+      .flat_map { _1.repetition? ? children_of(_1) : _1 }
+      .filter { _1.expression_reguliere? && _1.invalid_regexp? }
+      .each { |tdc| errors.add(:expression_reguliere, type_de_champ: tdc) }
+  end
+
```

Comment:
même remarque que mon commentaire de l'autre jour, ce serait bien de tout faire en une seule itération plutôt que passer par 4 tableaux intermédiaires assez illisibles